apps/analogquadclk/app.js

```diff
@@ -86,6 +86,21 @@ function draw() {
   lastModified = getHandBounds();
   //g.drawRect(lastModified); // debug
 }
+// Show launcher when middle button pressed
+Bangle.setUI({
+  mode: "clock",
+  remove: function() {
+    if (drawTimeout) clearTimeout(drawTimeout);
+    clockInfoMenuA.remove();
+    clockInfoMenuB.remove();
+    clockInfoMenuC.remove();
+    clockInfoMenuD.remove();
+    require("widget_utils").show(); // re-show widgets
```

Comment:
You might also want a:
```javascript
require("widget_utils").cleanup();
```

Reply:
Not aware of that one - will look!

Reply:
Should I just use the cleanup or both?

> Using Fast Loading? You'll have to re-display widgets with require("widget_utils").show() when your app unloads.

*https://www.espruino.com/Bangle.js+Hideable+Widgets*

Reply:
~~I think both, `show()` to bring the widgets back in, then cleanup to detach the `widget_utils` specific listeners, etc~~

Ignore me, I've just seen that `show()` calls `cleanup()`, doh

https://github.com/espruino/BangleApps/blob/f3b639f3e920fa74210b1aee2f9c62b16f21b61e/modules/widget_utils.js#L16-L18

But I did spot - `cleanup` deletes `Bangle.appRect`, which I think is a bug - the app we're fastloading into might rely on this for drawing, it should restore it instead. If that sounds right to you, I'll open a PR to fix it

Reply:
Thanks - but I think `delete Bangle.appRect; ` is fine. If you delete a built-in field then it doesn't really delete it - what it's doing is deleting the 'overwritten' appRect (because when you hide widgets the appRect has to be overwritten with one that includes the full screen)

Reply:
Ah nifty! Every day's a school day